app/master/slave_allocator.py

```diff
@@ -0,0 +1,57 @@
+from app.util.log import get_logger
+from app.util.ordered_set_queue import OrderedSetQueue
+from app.util.safe_thread import SafeThread
+
+
+class SlaveAllocator(object):
+    """
+    The SlaveAllocator class is responsible for allocating slaves to prepared builds.
+    """
+
+    def __init__(self, build_request_handler):
+        """
+        :type build_request_handler: BuildRequestHandler
+        """
+        self._logger = get_logger(__name__)
+        self._build_request_handler = build_request_handler
+        self._idle_slaves = OrderedSetQueue()
+        self._slave_allocation_worker_thread = SafeThread(
+            target=self._slave_allocation_loop, name='SlaveAllocationLoop', daemon=True)
+        self._slave_allocation_worker_thread.start()
```

Comment:
Do you think we should start the thread in a separate start() method instead of in the constructor? I know it doesn't make too much difference since we were (previously and currently) doing it in the ClusterMaster constructor anyway.